lib/puppet/property.rb

```diff
@@ -346,6 +362,29 @@ def insync?(is)
     end
   end
 
+  # This method tests if two values are insync? outside of the properties current
+  # should value. This works around the requirement for corrective_change analysis
+  # that requires two older values to be compared with the properties potentially
+  # custom insync? code.
+  #
+  # @param [Object] orig_should the original should value
```

Comment:
The phrasing of "original" here definitely shows its heritage from corrective_change. These should probably just be "should" and "is", and the documentation made more generic. This can be a post-merge cleanup, IMO